src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -22,12 +22,19 @@ DlgPrefWaveform::DlgPrefWaveform(QWidget* pParent, MixxxMainWindow* pMixxx,
     waveformOverviewComboBox->addItem(tr("HSV")); // "1"
     waveformOverviewComboBox->addItem(tr("RGB")); // "2"
 
-    // Populate waveform options.
+    // Get waveform names/types, not sorted alphabetically
     WaveformWidgetFactory* factory = WaveformWidgetFactory::instance();
-    QVector<WaveformWidgetAbstractHandle> handles = factory->getAvailableTypes();
-    for (int i = 0; i < handles.size(); ++i) {
-        waveformTypeComboBox->addItem(handles[i].getDisplayName(),
-                                      handles[i].getType());
+    m_handles = factory->getAvailableTypes();
```

Comment:
We can sort `m_handles` directly and do not need any extra members.

Even `m_handles` becomes obsolete if we store the type number directly as `userData` together at the combo box item.

Reply:
didn't figure hiw to sort a qvector last night. will look into it again. 

when pushing the rough, working state I was sure it can be boild down ; )

Reply:
https://en.cppreference.com/w/cpp/algorithm/sort

You need  to use a lamba for the custom comparison operator. Then just populate the combobox with name + type and throw away the temporary vector. The internal *model* of the combo box should be sufficient for storing all required information.

Reply:
coming from the widgetfactory I was thinking about lists only, which is total overkill here  :facepalm: 

that vector sorting with a custom compare function is very interesting.
Thanks for the model hint though: `waveformTypeComboBox->model()->sort(0);` would do exactly the same with way less effort.
It's just that [DlgPrefWaveform::slotSetWaveformType(int index)](https://github.com/mixxxdj/mixxx/blob/89493cc6f6a96f0a2db92bedc249ad3496450039/src/preferences/dialog/dlgprefwaveform.cpp#L242) passes the combobox index to [WaveformWidgetFactory::setWidgetTypeFromHandle](https://github.com/mixxxdj/mixxx/blob/89493cc6f6a96f0a2db92bedc249ad3496450039/src/waveform/waveformwidgetfactory.cpp#L493), 
but --until now-- we assume the combobox index matches the handles list index.

having diffulties to solve that...

Reply:
I just can't cast WaveformWidgetType::Type from QVariant. Well, I can but somewhere downstream it fails to update the renderer.

When I store the name and the `handles` index it works perfectly fine.
That's the current state.
Suggestions for improvements are welcome.